src/widget/translationAnimator.ts

```diff
@@ -169,11 +161,51 @@ export class TranslationAnimator extends Clutter.Actor {
         this.transitionContainer.ease(transitionConfig);
     }
 
+    setActors(actors: Clutter.Actor[]) {
+        this.transitionContainer.remove_all_children();
+        for (const actor of actors) {
+            reparentActor(actor, this.transitionContainer);
+        }
```

Comment:
This can lead to unnecessary reparenting, can it not?
We should only remove the children who are not part of the `actors` list.